### PR TITLE
Added test for empty middleware array

### PR DESCRIPTION
**Pull request by csarrazi**
This test is related to an untested edge case, mentioned by @Tobion in #99

### CODE REVIEW

File: src/Tests/DependencyInjection/CompilerPass/MiddlewarePassTest.php [hunk 1]
```diff
@@ -78,6 +78,21 @@ public function testMiddlewareWithPriority()
...
+
+        $container = $this->createContainer();
+        $container->setDefinition('client', $client);
+
+        $pass = new MiddlewarePass();
+        $pass->process($container);
+
+        $handler = $client->getArgument(0)['handler'];
+        $handlerDefinition = $container->getDefinition((string) $handler);
+        $this->assertCount(0, $calls = $handlerDefinition->getMethodCalls());
```
Comment (Tobion) [inline src/Tests/DependencyInjection/CompilerPass/MiddlewarePassTest.php]: `$calls` is unused

**Comment by Tobion** [comment]
Actually I would propose to not change the `handler` when no middleware is added. Currently it sets the default handler on the definition without any need.

Am I right that if you have a custom handler on your tagged guzzle client, this handler will be removed by your compiler pass? Should it not be wrapped to not lose ones own logic?

**Comment by csarrazi** [comment]
Actually, this is a reported issue. But you're right.

One of the main changes to do is basically move defining the handler stack into the extension class rather than the extension, and only create one using a DefinitionDecorator if the handler key is not set in the client's configuration.

In this case, the MiddlewarePass' role would be reduced to simply adding method calls defined handler stack.

**Comment by csarrazi** [comment]
Fixed in #104.
